### PR TITLE
[TF FE] Implement translators for TensorFlow ConvBackpropInput operations

### PR DESCRIPTION
**Pull request by rkazants**
**Details:** Previously, the translators did not support dynamic *input_sizes* input due to this TF FE was unable to convert PRNet model (from e2e testing). See the sub-graph with de-convolutional operation there:
image

Now the translators handle dynamic *input_sizes*, also it supports different padding modes including *EXPLICIT* mode, and the transformation preserves re-shapebility of a model.

PRNet model is successfully converted by TF FE and inferred.

**Ticket:** 88097

### CODE REVIEW

File: src/frontends/tensorflow/src/utils.cpp [hunk 1]
```diff
@@ -27,3 +28,20 @@ void ov::frontend::tensorflow::set_node_name(const std::string& node_name, const
 void ov::frontend::tensorflow::set_out_name(const std::string& out_name, const ov::Output<ov::Node>& output) {
     output.get_tensor().add_names({out_name});
 }
+
+ov::op::PadType convert_deconv_tf_padding(const ov::frontend::tensorflow::NodeContext& node, const std::string& tf_padding) {
```
Comment (itikhono) [inline src/frontends/tensorflow/src/utils.cpp:32]: I guess this method can be useful not only for deconv op but for all convolutions to get  auto_pad(padding_type) attribute. Please rename this.

Comment (rkazants) [reply src/frontends/tensorflow/src/utils.cpp:32]: Not at all, for Convolution operation SAME maps to SAME_UPPER. The different logic:)

File: src/frontends/tensorflow/src/op/conv_3d_backprop.cpp [hunk 2]
```diff
@@ -14,91 +14,95 @@ namespace tensorflow {
...
+    auto tf_explicit_paddings = node.get_attribute<std::vector<int64_t>>("explicit_paddings", {});
+    auto tf_data_format = node.get_attribute<std::string>("data_format", "NDHWC");
 
     TENSORFLOW_OP_VALIDATION(node,
                              tf_data_format == "NDHWC" || tf_data_format == "NCDHW",
-                             "Conv3DBackpropInputV2 data format is neither NDHWC nor NCDHW. "
-                             "Provided data format: ",
-                             tf_data_format);
+                             "Conv3DBackpropInput data format is neither NDHWC nor NCDHW");
+    TENSORFLOW_OP_VALIDATION(node,
```
Comment (itikhono) [inline src/frontends/tensorflow/src/op/conv_3d_backprop.cpp:35]: is it mistake when tf_explicit_paddings is set, but auto_pad type is not Explicit? or we can ignore these values?
maybe it's better to check `auto_pad == ov::op::PadType::EXPLICIT && tf_explicit_paddings.size() == 10`

File: src/frontends/tensorflow/src/utils.cpp (hunk 1 above)
Comment (itikhono) [reply src/frontends/tensorflow/src/utils.cpp:32]: ok, but we can check type of the operation here and return relevant value.
It looks like the function for convolutions will be duplicate most logic written here

File: src/frontends/tensorflow/src/op/conv_3d_backprop.cpp (hunk 2 above)
Comment (itikhono) [reply src/frontends/tensorflow/src/op/conv_3d_backprop.cpp:35]: let's say `auto_pad = Explicit`, so `auto_pad != ov::op::PadType::EXPLICIT` -> false
will it throw exception in this case? and it doesn't matter what tf_explicit_paddings.size() is actually is.

so maybe it's better:
if (auto_pad == ov::op::PadType::EXPLICIT)
    TENSORFLOW_OP_VALIDATION(..., tf_explicit_paddings.size() == 10, ...)

File: src/frontends/tensorflow/src/utils.cpp (hunk 1 above)
Comment (rkazants) [reply src/frontends/tensorflow/src/utils.cpp:32]: Ok, let us do this

File: src/frontends/tensorflow/src/op/conv_3d_backprop.cpp (hunk 2 above)
Comment (rkazants) [reply src/frontends/tensorflow/src/op/conv_3d_backprop.cpp:35]: Agree

File: src/frontends/tensorflow/src/openvino_conversions.hpp [hunk 3]
```diff
@@ -43,9 +43,9 @@ void convert_nchw_to_hw(const std::vector<T>& src, std::vector<size_t>& dst) {
 }
 }  // namespace detail
 
-void convert_nhwc_to_nchw(const std::string& op_name, bool need_convert, ov::Output<ov::Node>& ng_input);
+void convert_nhwc_to_nchw(bool need_convert, ov::Output<ov::Node>& input_);
```
Comment (mvafin) [inline src/frontends/tensorflow/src/openvino_conversions.hpp:46]: nit: ng_input was a better name why change?

**Review by mvafin — APPROVED** [review]
LGTM: but I would suggest considering merge 2d and 3d translators into a single translator function, using templates or something different, because code is very similar.

File: src/frontends/tensorflow/src/openvino_conversions.hpp (hunk 3 above)
Comment (rkazants) [reply src/frontends/tensorflow/src/openvino_conversions.hpp:46]: there is no longer NGraph term and it looks ugly, imho:)

Comment (mvafin) [reply src/frontends/tensorflow/src/openvino_conversions.hpp:46]: Why line 48 isn't changed then? Ending with underscore is pretty ugly too, imho :)

Comment (rkazants) [reply src/frontends/tensorflow/src/openvino_conversions.hpp:46]: Agree, I will do it:)

File: src/frontends/tensorflow/src/utils.cpp (hunk 1 above)
Comment (rkazants) [reply src/frontends/tensorflow/src/utils.cpp]: Fixed. Thanks.

File: src/frontends/tensorflow/src/op/conv_3d_backprop.cpp (hunk 2 above)
Comment (rkazants) [reply src/frontends/tensorflow/src/op/conv_3d_backprop.cpp]: Fixed. Thanks.

File: src/frontends/tensorflow/src/openvino_conversions.hpp (hunk 3 above)
Comment (rkazants) [reply src/frontends/tensorflow/src/openvino_conversions.hpp]: Fixed. Thanks.

File: src/frontends/tensorflow/src/op/conv_2d_backprop.cpp [hunk 4]
```diff
@@ -14,86 +16,99 @@ namespace tensorflow {
 namespace op {
 
 OutputVector translate_conv_2d_backprop_input_op(const NodeContext& node) {
-    auto ng_filter = node.get_input(1), ng_out_backprop = node.get_input(2);
+    TENSORFLOW_OP_VALIDATION(node, node.get_input_size() >= 2, "Conv2DBackpropInput must have at least three inputs.");
```
Comment (itikhono) [inline src/frontends/tensorflow/src/op/conv_2d_backprop.cpp:19]: at least three inputs, but why >= 2?
should be > 2 or >= 3?

File: src/frontends/tensorflow/src/op/conv_3d_backprop.cpp [hunk 5]
```diff
@@ -14,91 +14,101 @@ namespace tensorflow {
 namespace op {
 
 OutputVector translate_conv_3d_backprop_input_v2_op(const NodeContext& node) {
-    auto ng_filter = node.get_input(1);
-    auto ng_out_backprop = node.get_input(2);
+    TENSORFLOW_OP_VALIDATION(node, node.get_input_size() >= 2, "Conv3DBackpropInput must have at least three inputs.");
```
Comment (itikhono) [inline src/frontends/tensorflow/src/op/conv_3d_backprop.cpp:17]: at least three inputs, but why >= 2?
should be > 2 or >= 3?

**Review by itikhono — APPROVED** [review]

File: src/frontends/tensorflow/src/op/conv_2d_backprop.cpp (hunk 4 above)
Comment (rkazants) [reply src/frontends/tensorflow/src/op/conv_2d_backprop.cpp]: Great catch! Fixed.

File: src/frontends/tensorflow/src/op/conv_3d_backprop.cpp (hunk 5 above)
Comment (rkazants) [reply src/frontends/tensorflow/src/op/conv_3d_backprop.cpp]: Great catch! Fixed.

**Pull request merged**